File: 3.1C/java/com/example/quizapp/MainActivity.java
package com.example.quizapp;

import android.content.Intent;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        EditText name = findViewById(R.id.name);
        if (getIntent().getStringExtra("Name") != null) {
            name.setText(getIntent().getStringExtra("Name"));
        }
        findViewById(R.id.startQuizbutton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(v.getContext(), QuizActivity.class);
                intent.putExtra("Name", name.getText().toString());
                startActivity(intent);
            }
        });
    }
}

File: 5.1C News Code/java/com/example/news/MainActivity.kt
package com.example.news
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupActionBarWithNavController

private const val TAG="MainActivity"
class MainActivity : AppCompatActivity() {
    val TAG = "MainActivity"
    private lateinit var navController: NavController


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val navHostManager = supportFragmentManager.findFragmentById(R.id.fragmentContainerView) as NavHostFragment
        populateArticles()
        navController = navHostManager.navController
        Log.i("MainActivity", navController.toString())
        setupActionBarWithNavController(navController)

    }

    private fun populateArticles(){
        val art1 = Articles("Mt Everest Report", R.drawable.everest,"Mt Everest 10 people climbed today!")
        articles.add(art1)

        val art2 = Articles("Bogo Sort Analysis", R.drawable.bogo,"Bogosort the best")
        articles.add(art2)

        val art3 = Articles("Nukes?", R.drawable.boom,"Nukes!!")
        articles.add(art3)

        val art4 = Articles("Obama vs Trump Boxing?", R.drawable.obvt,"Obama is allegedly fighting Donald Trump in a boxing fight")
        articles.add(art4)
    }

    override fun onSupportNavigateUp(): Boolean {
        return navController.navigateUp() || super.onSupportNavigateUp()

    }

}